Node.js/GettingStarted/azure_batch_client.js

```diff
@@ -0,0 +1,196 @@
+var poolConfig = {id:poolid, displayName:poolid,vmSize:vmSize,virtualMachineConfiguration:vmconfig,targetDedicated:numVms,enableAutoScale:false }
+var poolAddOptionsConfig = {clientRequestId:clientPoolRequestID,returnClientRequestId:true}
+var poolOptions = {poolAddOptions:poolAddOptionsConfig}
+var pool = batch_client.pool.add(poolConfig,poolOptions,function(error,result,request,response){            
+    //        console.log("Pool add response stream" + response.body);
+            if(error != null)
+            {
+                console.log(error.response);
+            }
+            
+        });
+
+console.log("Submitted request to create pool...");
+
+
+var isPoolStateActive = false;
+var poolCreationError = false;
+var maxRetry = 1000;
+var numRetries = 0;
+var jobCreated = false;
+
+while(isPoolStateActive == false && poolCreationError == false && numRetries < maxRetry)
+{
+    
+    var cloudPool = batch_client.pool.get(poolid,function(error,result,request,response){
+        if(error == null)
+        {
+            if(result.state == "active")
+            {
+                  isPoolStateActive = true;
+                  if(jobCreated==true)
+                  {
+                      // do nothing
+                  }
+                  else
+                  {
+                     jobCreated = true;
+                     console.log("Pool created successfully now submitting job");
+                     createJob();
+                  }
+                  //// Create the Job 
+            }
+        }
+        else
+        {
+            if(error.statusCode==404)
+            {
+                //console.log("Pool not found yet returned 404...");    
+                           
+            }
+            else
+            {
+                poolCreationError = true;
+            }
+        }
+        });
+  
+  numRetries++;
+ 
+}
+
+console.log("Pool Status : " + isPoolStateActive);
+
+
+
+
+
+
+function createJob()
+{
+    console.log("Inside create job...");
+    var sh_url = "Shell script SAS URI";
+    var job_prep_task_config = {id:"installprereq",commandLine:"sudo sh startup_prereq.sh > startup.log",resourceFiles:[{'blobSource':sh_url,'filePath':'startup_prereq.sh'}],waitForSuccess:true,runElevated:true}
+    // Setting up Batch pool configuration
+    var poolConfigJob = {poolId:poolid}
+    
+    
+    var job_config = {id:jobId,displayName:"process csv files",jobPreparationTask:job_prep_task_config,poolInfo:poolConfigJob}
+    
+     // Adding Azure batch job to the pool
+     var job = batch_client.job.add(job_config,function(error,result){
+        if(error !=null)
+        {
+            console.log(error);
+        }        
+        
+    });       
+    var maxRetryJob = 100;
+    var isJobCreated = false
+    var isJobCreatedError = false
+    var numRetriesJob = 0;
+    var taskSubmitted = false;
+    while(numRetriesJob < maxRetryJob && isJobCreated == false && isJobCreatedError == false)
```

Comment:
This seems a bit odd.  If `job.add` succeeded then surely the job is created.  Why do we need to poll for its state?